### PR TITLE
Implement CORS support in the s3 module

### PR DESCRIPTION
**Pull request by legalosLOTR**
# Description
Pretty straightforward
TODO: update docs

# Test plan
Try creating a s3 bucket via opta, with various permutations of the cors_rule and make sure it works

### CODE REVIEW

File: config/tf_modules/aws-s3/variables.tf [hunk 1]
```diff
@@ -27,3 +27,7 @@ variable "bucket_policy" {
   default = null
 }
 
+variable "cors_rule" {
+  type = any
+  default = null
```
Comment (juandiegopalomino) [inline config/tf_modules/aws-s3/variables.tf:32]: why not default = []?

File: config/schema/modules/aws-s3.yaml [hunk 2]
```diff
@@ -3,6 +3,12 @@ name: str(required=False)
 env_name: str(required=False)
 layer_name: str(required=False)
 module_name: str(required=False)
-bucket_name: str(required=False)
+bucket_name: str(required=True)
 block_public: any(required=False)
-bucket_policy: any(required=False)
\ No newline at end of file
+bucket_policy: any(required=False)
+cors_rule:
```
Comment (juandiegopalomino) [inline config/schema/modules/aws-s3.yaml:9]: is this required=False by default?

File: config/tf_modules/aws-s3/variables.tf (hunk 1 above)
Comment (legalosLOTR) [reply config/tf_modules/aws-s3/variables.tf:32]: It's a map, not a list

File: config/schema/modules/aws-s3.yaml (hunk 2 above)
Comment (legalosLOTR) [reply config/schema/modules/aws-s3.yaml:9]: Ah it's not! Let me figure out a way to fix it

File: config/tf_modules/aws-s3/bucket.tf [hunk 3]
```diff
@@ -8,6 +8,17 @@ resource "aws_s3_bucket" "bucket" {
     }
   }
   force_destroy = true
+
+  dynamic "cors_rule" {
+    for_each = [var.cors_rule]
```
Comment (juandiegopalomino) [inline config/tf_modules/aws-s3/bucket.tf:13]: for_each = var.cors_rule == null ? [] : [var.cors_rule]

**Review by juandiegopalomino — APPROVED** [review]

**Pull request merged**